src/main.cpp

```diff
@@ -1057,6 +1057,7 @@ class ATTRIBUTE_HIDDEN SampleReader
   virtual void AddStreamType(INPUTSTREAM_TYPE type, uint32_t sid){};
   virtual void SetStreamType(INPUTSTREAM_TYPE type, uint32_t sid){};
   virtual bool RemoveStreamType(INPUTSTREAM_TYPE type) { return true; };
+  virtual bool IsStarted() const { return true; };
```

Comment:
Why is the default true?

Reply:
I set it up like this so the other stream readers don't go through the `StartReader` block as there's no issue with how they work.
Having said that I can see now that name of the function isn't true to how it behaves, it should really return `m_Started`, correct? And then we either check the stream type as well i.e (pseudo code) `if (reader == fmp4 && !timing_stream_->reader_->IsStarted()) ...` or if possible even better to implement IsStarted() for the other readers.

Reply:
Yes, much better!

Reply:
Ok I hope we're good now.
No default implementation for `IsStarted`. 
Implementations for all sample readers.
Subtitle reader always returns true - there is no `m_started` that we keep track of in this reader.
bReset is also initialized now to `true`. There is a code path in `SeekTime` where bReset doesn't get modified so that's why I went with true.